setup.py

```diff
@@ -1,23 +1,149 @@
 # See: https://stackoverflow.com/a/49684835
-with open('requirements_common.txt') as f:
+with open(path.join(here, 'requirements.txt'), encoding='utf-8') as f:
     requirements = f.readlines()
 
-# Get README
-this_directory = path.abspath(path.dirname(__file__))
-with open(path.join(this_directory, 'README.md'), encoding='utf-8') as f:
-    long_description = f.read()
 
-# Get Release version
-path_version = path.join(this_directory, 'ivadomed', 'version.txt')
-with open(path_version) as f:
-    version = f.read().strip()
+def get_whl_info():
+    """Returns the whl info to handpick torch whls.
+
+    whl format: {dist}-{version}(-{build})?-{python_tag}-{abi_tag}-{platform}.whl
+
+    Returns:
+        str, str, str: python_tag, abi_tag, platform.
+    """
+    py_version = python_version()[0:3].replace('.', '')
+    python_tag = f'cp{py_version}'
+
+    if py_version == '36' or py_version == '37':
+        abi_tag = f'{python_tag}m'
+    elif py_version == '38' or py_version == '39':
+        abi_tag = python_tag
+
+    sysinfo = uname()
+    if sysinfo.system == 'Linux':
+        if sysinfo.machine == 'x86_64':
+            platform = 'linux_x86_64'
+            return python_tag, abi_tag, platform
+        elif sysinfo.machine == 'aarch64':
+            raise ValueError(f"ivadomed doesn't handpick torch whls for {sysinfo.machine}, \
+            install torch==1.8.0+cpu, torchvision==0.9.0+cpu or \
+            torch==1.8.1+cu111, torchvision==0.9.1+cu111 from \
+            https://pytorch.org/get-started/previous-versions/")
+    elif sysinfo.system == 'Windows' and sysinfo.machine == 'AMD64':
+        platform = 'win_amd64'
+        return python_tag, abi_tag, platform
+    elif sysinfo.system == 'Darwin':
+        # TODO: update mac conditions as we upgrade to higher torch versions
+        # with separate binaries for 10 and 11 versions   
+        # macinfo = mac_ver()
+        # if macinfo[0].startswith('10') and macinfo[2] == 'x86_64':
+        platform = 'macosx_10_9_x86_64'
+        return python_tag, abi_tag, platform 
 
 
+def get_torch_whls(backend):
+    """Handpicks whls for torch and torchvision based on the backend.
+
+    Args:
+        backend (str): cpu or gpu
+
+    Returns:
+        list[str]: List of handpicked torch et al. whls
+    """
+    python_tag, abi_tag, platform = get_whl_info()
+    if backend == 'cpu':
+        if platform.startswith('macosx'):
+            _torch_whl = f'-{python_tag}-none-{platform}.whl'
+            torch_whl = f'torch@https://download.pytorch.org/whl/cpu/torch-1.8.0{_torch_whl}'
+            _torchvision_whl = f'-{python_tag}-{abi_tag}-{platform}.whl'
+            torchvision_whl = f'torchvision@https://download.pytorch.org/whl/torchvision-0.9.0{_torchvision_whl}'
+            return torch_whl, torchvision_whl
+        else:
+            whl = f'%2B{backend}-{python_tag}-{abi_tag}-{platform}.whl'
+            torch_whl = f'torch@https://download.pytorch.org/whl/cpu/torch-1.8.0{whl}'
```

Comment:
would this work to get this uploadable to pypi?

```suggestion
            torch_whl = f'torch==1.8.0+cpu'
```


And then instruct users:

`pip install --find-links https://download.pytorch.org/whl/torch ivadomed`

?

Reply:
And sct would say:

```
pip install --find-links https://download.pytorch.org/whl/torch spinalcordtoolbox
```